modules/identityserver/src/Volo.Abp.IdentityServer.Domain/Volo/Abp/IdentityServer/AbpIdentityServerDomainModule.cs

```diff
@@ -110,6 +105,37 @@ private static void AddIdentityServer(IServiceCollection services)
         }
     }
 
+    private static IIdentityServerBuilder AddIdentityServer(IServiceCollection services, AbpIdentityServerBuilderOptions abpIdentityServerBuilderOptions)
+    {
+        services.Configure<IdentityServerOptions>(options =>
+        {
+            options.Events.RaiseErrorEvents = true;
+            options.Events.RaiseInformationEvents = true;
+            options.Events.RaiseFailureEvents = true;
+            options.Events.RaiseSuccessEvents = true;
+        });
+
+        var identityServerBuilder = services.AddIdentityServerBuilder()
+            .AddRequiredPlatformServices()
+            .AddCoreServices()
+            .AddDefaultEndpoints()
+            .AddPluggableServices()
+            .AddValidators()
+            .AddResponseGenerators()
+            .AddDefaultSecretParsers()
+            .AddDefaultSecretValidators();
+
+        if (abpIdentityServerBuilderOptions.AddIdentityServerCookieAuthentication)
+        {
+            identityServerBuilder.AddCookieAuthentication();
+        }
```

Comment:
I am not sure if it is a good idea to add cookie authentication options at the **Domain** layer.

Reply:
Identity Server also will do this by default.

https://github.com/IdentityServer/IdentityServer4/blob/3ff3b46698f48f164ab1b54d124125d63439f9d0/src/IdentityServer4/src/Configuration/DependencyInjection/IdentityServerServiceCollectionExtensions.cs#L40